compiler/src/dotty/tools/dotc/typer/Inferencing.scala

```diff
@@ -206,7 +206,11 @@ object Inferencing {
     }
 
     val widePt = if (ctx.scala2Mode || refinementIsInvariant(tp)) pt else widenVariantParams(pt)
-    tp <:< widePt
+    import dotty.tools.dotc.config.Printers.gadts
+    gadts.println(i"constraining pattern type $tp <:< $widePt")
+    val res = tp <:< widePt
```

Comment:
This looks like a good application of `trace`. Benefit: it keeps track of indentation.